src/pages/PostDetailPage.jsx

```diff
@@ -1,32 +1,67 @@
-import React from 'react';
+import Swal from 'sweetalert2';
 import { MdOutlineLocationOn, MdLocationPin } from 'react-icons/md';
 import { BsArrowDown } from 'react-icons/bs';
+import { useState } from 'react';
 import Button from '../components/atoms/Button';
 import OtherNav from '../components/atoms/OtherNav';
 
 const PostDetailPage = () => {
-  return (
-    <div className="page--layout">
-      {/* 파란색 부분 */}
-      <div className="bg-sky-blue h-60 rounded-b-3xl">
-        <OtherNav iconColor="#fff" bgColor="#000" />
-        <div className="p-5">
-          <div className="mt-1 text-white text-xl">매칭을 기다리고 있어요.</div>
-          <div className="p-4">
-            <div className="flex items-center">
-              <MdOutlineLocationOn className="mr-1" size="30" style={{ color: 'white' }} />
-              <span className="text-white">스타벅스 전대후문점</span>
-            </div>
-            <BsArrowDown className="ml-2 my-1" style={{ color: 'white' }} />
-            <div className="flex items-center">
-              <MdLocationPin className="mr-1" size="30" style={{ color: 'white' }} />
-              <span className="text-white">공과대학 7호관</span>
-            </div>
-          </div>
-        </div>
-      </div>
-      {/* 하얀색 부분 */}
-      <div className="px-8">
+  const [page, setPage] = useState(0);
+  const [isMatch, setIsMatch] = useState(false);
+
+  // 이 음료 픽업하기 버튼을 눌렀을 때 뜨는 모달창
+  const pickUpBtnModal = () => {
+    return Swal.fire({
+      title: '이 음료를 픽업하시겠습니까?',
+      showCancelButton: true,
+      cancelButtonText: '취소',
+      confirmButtonText: '수락',
+      confirmButtonColor: '#0075ff',
+      heightAuto: true,
+    }).then((result) => {
+      if (result.isConfirmed) {
+        setPage(1);
+      }
+    });
+  };
+
+  // 픽업하기 눌렀다가 취소할 때
+  const cancel = () => {
+    setPage(0);
+  };
+
+  // 입력완료 버튼
+  const doneWrite = () => {
+    setPage(0);
+    setIsMatch(true);
+    return Swal.fire({
+      icon: 'success',
+      title: '매칭이 완료 되었어요!',
+      showConfirmButton: false,
+      timer: 3000,
+    });
+  };
+
+  // 픽업버튼 활성화할지 안할지
+  const pickUpBtn = (isMatched) => {
+    if (isMatched) {
+      return (
+        <Button bgColor="bg-zinc-300" disabled="disable">
+          이미 매칭된 공고
+        </Button>
+      );
+    }
+    return (
+      <Button onClick={pickUpBtnModal} bgColor="bg-blue">
+        이 음료 픽업하기
+      </Button>
+    );
+  };
```

Comment:
```suggestion
  const pickUpBtn = (isMatched) => {
    if (isMatched) {
      return (
        <Button bgColor="bg-zinc-300" disabled>
          이미 매칭된 공고
        </Button>
      );
    };
    else {
      return (
         <Button onClick={pickUpBtnModal} bgColor="bg-blue">
            이 음료 픽업하기
          </Button>
        );
     };
  };
```

`else`를 명시해주는 게 더 명확한 의미 전달이 되지 않을까..! 하는 생각입니다..!!!